org.mwc.cmap.legacy/src/MWC/GUI/Shapes/Symbols/SVG/SVGRectangle.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *    Debrief - the Open Source Maritime Analysis Application
+ *    http://debrief.info
+ *
+ *    (C) 2000-2018, Deep Blue C Technology Ltd
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the Eclipse Public License v1.0
+ *    (http://www.eclipse.org/legal/epl-v10.html)
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ */
+package MWC.GUI.Shapes.Symbols.SVG;
+
+import java.awt.Color;
+
+import org.jsoup.nodes.Element;
+
+import MWC.GUI.CanvasType;
+
+public class SVGRectangle extends SVGElement
+{
+  
+  public SVGRectangle(Element dom)
+  {
+    super(dom);
+  }
+
+  @Override
+  public void render(CanvasType dest, double sym_size,
+      java.awt.Point origin_coords, double rotation_degs)
+  {
+    final double x = Double.parseDouble(get_dom().attr("x")) * sym_size;
```

Comment:
Minor point. From your implementation, I see that SVGRectangle only gets created once we're already in a paint process `SVGShape.paint()`.

So, I believe we should extract x, y, width, height in the constructor for SVGRectangle.  Then we know they'll just be called once.

Reply:
I totally agree with you!! I am doing the changes mentioned